core/src/main/java/io/grpc/internal/AbstractServerImplBuilder.java

```diff
@@ -116,6 +126,14 @@ public final T addService(BindableService bindableService) {
   }
 
```

Comment:
ServerServiceDefinition is a final class. We need to check for instances of the InternalNotifyOnServerBuild interface. Can an instance of a final class also implement an interface?